### DISCUSSION TITLE
Qucs EDD project altered to work with Qucs-S doesn't work anymore

### DISCUSSION THREAD

**Discussion by tomhajjar** [ra3xdh/qucs_s#579] [Q&A]
Vadim a Qucs MESFET example using an EDD that you fixed in the following thread doesn't work anymore.
https://github.com/ra3xdh/qucs_s/issues/152

The files work under Qucs-S 0.0.24 and any ngspice version. However the exported netlists from Qucs-S 0.0.24 don't work using DuSpice. I have NEVER seen this happen before. VERY strange...

The files fail under Qucs-S 24.1.0 regardless of ngspice version. 

The netlists have a slight variation between Qucs-S 0.0.24 and Qucs-S 24.1.0. One uses a small "v" versus a capital "V".
"alter vgs = $vgs_act" versus alter "Vgs = $vgs_act".

Whatever Qucs-S 0.0.24 does differently "interacting" with ngspice allows it to work while  Qucs-S 24.1.0 doesn't. I altered the two netlists so ngspice would create different error logs. Both generate the identical errors. Again why Qucs-S 0.0.24 works is baffling.

I started this because MESFETs.lib doesn't work and was going to make a version using EDD.

Qucs-S 0.0.24 is difficult to use unless you redo all subcircuit Headers to be <Qucs Schematic 0.0.24>. You also have to redo the simulator configuration each time you change. Qucs-S 0.0.24 always defaults to xyce and Qucs-S 24.1.0 defaults to ngspice. 

![2024-01-21_144720](https://github.com/ra3xdh/qucs_s/assets/102838604/e94d854d-fb68-4259-9b7c-86a181332f64)
![2024-01-21_144659](https://github.com/ra3xdh/qucs_s/assets/102838604/1d150ea8-beff-42cc-9abb-7de42304129d)

**Comment by ra3xdh** [comment]
I cannot reproduce the attached error logs. The MESFET models work on my machine as expected. Moving this issue to discussion: https://github.com/ra3xdh/qucs_s/discussions

**Comment by tomhajjar** [comment]
Is this the same issue I had in #569? .PARAM values won't be available in later equations. 

![2024-02-17_133035](https://github.com/ra3xdh/qucs_s/assets/102838604/47eb8a3e-b442-400b-b402-21c2e06046ac)

**Comment by ra3xdh** [comment]
Vt is defined before TK in the provided netlist. TK is undefined in the Vt equation. Try to change the equations order. I am wondering why my Ngspice build accepts this.

**Comment by tomhajjar** [comment]
It works with Qucs-S 0.0.24 and every version of ngspice I have.

**Comment by ra3xdh** [comment]
The error in the log comes from Ngspice, not from Qucs-S. I cannot reproduce it using my build.

**Comment by tomhajjar** [comment]
The netlists are different. I also attached the 0.0.24 and 24.1 project folders. 0.0.24 won't load newer rev sub-circuits so best to make two folders.

**Comment by ra3xdh** [comment]
The only difference is the `Tc` parameter of the resistors. This cannot caused simulation error.

**Comment by tomhajjar** [comment]
The problem is my default .spiceinit = "set ngbehavior=ltpsa" in my Windows user directory.

Something is different between 0.0.24 and 24.1. I have used the same default .spiceinit long before 0.0.24.

I ran into a similar issue in the past. Qucs-S uses the spice4qucs folder to write the .spiceinit. Qucs-S writes and deletes the .spiceinit every simulation run. Holger explained where ngspice looks for the .spiceinit.

Does 24.1 handle .spiceinit differently than 0.0.24? Will I need a .spiceinit in every schematic and delete my default one? If so it needs to be added as a prerequisite.

https://sourceforge.net/p/ngspice/discussion/120973/thread/afefdb21be/

[MESFET_Curtice_EDD_prj.zip](https://github.com/ra3xdh/qucs_s/files/14320493/MESFET_Curtice_EDD_prj.zip)
